src/chat.rs

```diff
@@ -1898,15 +1916,46 @@ pub fn get_chat_id_by_grpid(context: &Context, grpid: impl AsRef<str>) -> (u32,
         .unwrap_or((0, false, Blocked::Not))
 }
 
-pub fn add_device_msg(context: &Context, chat_id: u32, text: impl AsRef<str>) {
+pub fn add_device_msg(context: &Context, msg: &mut Message) -> Result<MsgId, Error> {
+    let (chat_id, _blocked) =
+        create_or_lookup_by_contact_id(context, DC_CONTACT_ID_DEVICE, Blocked::Not)?;
+    let rfc724_mid = dc_create_outgoing_rfc724_mid(None, "@device");
+
+    prepare_msg_blob(context, msg)?;
+    unarchive(context, chat_id)?;
+
+    context.sql.execute(
```

Comment:
why is this a custom insert?  I appreciate you don't want to call send_msg() but i'm surprised to see a whole new sql query.  I probably missed something though.

Reply:
there is just no other `INSERT INTO msgs` that would _only_ do an insert. the ones in receive_imf(), set_draft(), set_info_msg() do other, unwanted stuff and/or have a customized insert on their own.

but i think this is no issue. these sqls are easy enough and so it is pretty clear what happens.